versioning.mk

```diff
@@ -38,6 +38,7 @@ VERSION_LDFLAGS = -X github.com/vmware-tanzu/antrea/pkg/version.Version=$(VERSIO
 VERSION_LDFLAGS += -X github.com/vmware-tanzu/antrea/pkg/version.GitSHA=$(GIT_SHA)
 VERSION_LDFLAGS += -X github.com/vmware-tanzu/antrea/pkg/version.GitTreeState=$(GIT_TREE_STATE)
 VERSION_LDFLAGS += -X github.com/vmware-tanzu/antrea/pkg/version.ReleaseStatus=$(RELEASE_STATUS)
+VERSION_LDFLAGS += -X k8s.io/component-base/version.gitVersion=$(VERSION)
```

Comment:
I didn't see Antrea's code read this variable, why it needs to set it?

Reply:
This is useful for initializing `legacyregistry.defaultRegistry`. Prometheus metrics handler will add deprecated descriptions or hide the metrics entirely according to the metrics deprecation lifecycle.

Reply:
I am not a big fan of this. It feels wrong to rely on a build variable from K8s.
Can we stop relying on the legacyregistry and instead instantiate our own with `NewKubeRegistry`? We can overwrite the `BuildVersion` function with something that returns the Antrea version.

Reply:
I wonder whether this is necessary. I checked and find there are some other packages of apiserver rely on legacyregistry directly. e.g. https://github.com/kubernetes/apiserver/blob/master/pkg/server/routes/metrics.go#L47-L51
If we use our own instance of `KubeRegistry ` we might lose these generic k8s API server metrics.

Reply:
So our only choices are 1) inject our own gitVersion and we may mess with existing k8s metrics (don't they also rely on the version to decide whether a metric is deprecated / should be hidden?) or 2) do not inject our own gitVersion and lose support for automatic deprecation warnings / hidden metrics?

@srikartati I think you already looked into this when you worked on https://github.com/vmware-tanzu/antrea/pull/1217. Do you have any insight?

Reply:
AFAIK there are no deprecated versions set for these existing k8s metrics. And if we do not set gitVersion in `component-base` it would be defaulting to `v0.0.0`.